### PR TITLE
Don't save dupe entries in Extra Folders to prevent performance degradation

### PR DESCRIPTION
**Pull request by unab0mb**
…dation, fix dupe direct/music folders on 1st run, ensure direct/music always in configuration per EQ2Alyza's guide.

### CODE REVIEW

File: 7thWorkshop/fSettings.cs [hunk 1]
```diff
@@ -105,7 +105,17 @@ public partial class fSettings : Form {
         private void bOK_Click(object sender, EventArgs e) {
 
             Sys.Settings.SubscribedUrls = txtSubscriptions.Lines.ToList();
-            if (txtExtraFolders.Lines.ToList().Count > 0 ) Sys.Settings.ExtraFolders = txtExtraFolders.Lines.ToList();
+
+            //Convert Extra Folders textbox to a list for comparison and editing
+            List<string> ExtraFoldersList = txtExtraFolders.Lines.ToList();
+
+            //Make sure that the direct and music folders are always present
+            if (!ExtraFoldersList.Contains("direct", StringComparer.OrdinalIgnoreCase)) ExtraFoldersList.Add("direct");
```
Comment (julianxhokaxhiu) [inline 7thWorkshop/fSettings.cs]: Please use `InvariantCultureIgnoreCase`

File: 7thWorkshop/fSettings.cs [hunk 2]
```diff
@@ -105,7 +105,17 @@ public partial class fSettings : Form {
...
+            if (!ExtraFoldersList.Contains("music")) ExtraFoldersList.Add("music");
```
Comment (julianxhokaxhiu) [inline 7thWorkshop/fSettings.cs]: Here it is missing, please add the `StringComparer.InvariantCultureIgnoreCase` as well

File: 7thWorkshop/fSettings.cs [hunk 3]
```diff
@@ -105,7 +105,17 @@ public partial class fSettings : Form {
...
+
+            //Save the Extra Folders and remove any duplicate entries for best performance
+            Sys.Settings.ExtraFolders = ExtraFoldersList.Distinct(StringComparer.OrdinalIgnoreCase).ToList();
```
Comment (julianxhokaxhiu) [inline 7thWorkshop/fSettings.cs]: Same here, please use `InvariantCultureIgnoreCase`

File: 7thHeaven.Code/Sys.cs [hunk 4]
```diff
@@ -138,7 +138,8 @@ public static class Sys {
                 Settings.Options = GeneralOptions.AutoImportMods;
             }
 
-            if (Settings.ExtraFolders == null || Settings.ExtraFolders.Count == 0) Settings.ExtraFolders = new List<string>() { "direct", "music" };
+            //Fixes dupe direct/music folders on 1st run.
+            //if (Settings.ExtraFolders == null || Settings.ExtraFolders.Count == 0) Settings.ExtraFolders = new List<string>() { "direct", "music" };
```
Comment (julianxhokaxhiu) [inline 7thHeaven.Code/Sys.cs]: Just delete both lines, if we need this code we can get it back from git

**Review by julianxhokaxhiu — CHANGES_REQUESTED** [review]
Overall the PR looks good but there are minor fixes required. Let me know when they are done by requesting another review.

**Review by julianxhokaxhiu — APPROVED** [review]
LGTM

**Comment by TurBoss** [comment]
just tested and works fine

Thank you!

**Pull request merged**